### PR TITLE
Add PHP 8.0 support

### LINKED ISSUE #13: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by bfoosness**
|    Q          |   A
|-------------- | ------
| New Feature   | yes

### Description

Close #13

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,41 +12,26 @@ env:
...
-    - php: 7.2
+    - php: 7.4
       env:
         - DEPS=lowest
-    - php: 7.2
+    - php: 7.4
       env:
         - DEPS=latest
-    - php: 7.3
+    - php: 'nightly'
```
Comment (snapshotpl) [inline .travis.yml]: `8.0` is supported now

**Review by snapshotpl — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -12,41 +12,26 @@ env:
 matrix:
   fast_finish: true
   include:
-    - php: 5.6
```
Comment (glensc) [inline .travis.yml]: "Update .travis.yml" is not much useful commit message. seems like clickbait.

**Review by Slamdunk — APPROVED** [review]

**Comment by zluiten** [comment]
What is the status of this?

**Comment by bfoosness** [comment]
> What is the status of this?

As far as I know it's done, and it was previously approved. I'm just waiting on either feedback or a merge.

**Pull request merged**